examples/40_advanced/example_resampling.py

```diff
@@ -98,8 +98,9 @@
 # data by the first feature. In practice, one would use a splitting according
 # to the use case at hand.
 
+selected_indices = (X_train[:, 0] < np.mean(X_train[:, 0])).astype(int)
```

Comment:
Is the `astype(int)` necessary here? If we cast to int we would only select the first and the second data point, wouldn't we?

Reply:
[PredefinedSplit](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.PredefinedSplit.html#sklearn-model-selection-predefinedsplit) must have `len(x) == len(selected_indices)` where `selected_indices[i]` is an int specifying which split it belongs to.

```python
In [7]: x = np.array([
   ...:     [1,2,3],
   ...:     [4,5,6],
   ...:     [7,8,9],
   ...: ])
   
In [8]: x[:,0]
Out[8]: array([1, 4, 7])

In [9]: x[:,0] <= np.mean(x[:,0])
Out[9]: array([ True,  True, False])

In [10]: (x[:,0] <= np.mean(x[:,0])).astype(int)
Out[10]: array([1, 1, 0])  # sample 0, 1 are test, sample 2 is train
```

Reply:
Thanks for explaining, I clearly misunderstood the API of that splitter.